Source/Editor/Windows/Assets/JsonAssetWindow.cs

```diff
@@ -16,6 +19,53 @@ namespace FlaxEditor.Windows.Assets
     /// <seealso cref="FlaxEditor.Windows.Assets.AssetEditorWindow" />
     public sealed class JsonAssetWindow : AssetEditorWindowBase<JsonAsset>
     {
+        private class ObjectPasteUndo : IUndoAction
+        {
+            /// <inheritdoc />
+            public string ActionString => "Object Paste Undo";
+
+            private JsonAssetWindow _window;
+            private object _oldObject;
```

Comment:
`_oldObject` and `_newObject` should be stored as `string` and use Json serialization to store in memory so it will properly work when code is hot-reloaded in Editor.